### ISSUE TITLE
[MPIR-363] Upgrade dependencies to latest versions

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-project-info-reports-plugin#215]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** opened **[MPIR-363](https://issues.apache.org/jira/browse/MPIR-363?redirect=false)** and commented

```
The following plugin updates are available:
maven-changes-plugin ............................... 2.11 -> 2.12.1
maven-jarsigner-plugin ............................... 1.3.2 -> 1.4
org.codehaus.mojo:keytool-maven-plugin ................. 1.0 -> 1.5

The following dependencies in Dependency Management have newer versions:
org.apache.maven.plugin-tools:maven-plugin-annotations ... 3.4 -> 3.5.1
org.codehaus.plexus:plexus-component-annotations ........ 1.6 -> 1.7.1

The following dependencies in Dependencies have newer versions:
commons-validator:commons-validator ..................... 1.5.1 -> 1.6
httpunit:httpunit ....................................... 1.6 -> 1.6.1
junit:junit ............................................ 3.8.2 -> 4.12
org.apache.maven:maven-artifact ....................... 2.2.1 -> 3.5.3
org.apache.maven:maven-model .......................... 2.2.1 -> 3.5.3
org.apache.maven:maven-plugin-api ..................... 2.2.1 -> 3.5.3
org.apache.maven:maven-project .................. 2.2.1 -> 3.0-alpha-2
org.apache.maven:maven-repository-metadata ............ 2.2.1 -> 3.5.3
org.apache.maven:maven-settings ....................... 2.2.1 -> 3.5.3
org.apache.maven.doxia:doxia-core ......................... 1.6 -> 1.8
org.apache.maven.doxia:doxia-decoration-model ............. 1.6 -> 1.8
org.apache.maven.doxia:doxia-integration-tools ............ 1.6 -> 1.8
org.apache.maven.doxia:doxia-sink-api ..................... 1.6 -> 1.8
org.apache.maven.doxia:doxia-site-renderer ................ 1.6 -> 1.8
org.apache.maven.plugin-testing:maven-plugin-testing-harness ... 1.3 -> 3.3.0
org.apache.maven.reporting:maven-reporting-impl ......... 2.4 -> 3.0.0
org.apache.maven.shared:maven-common-artifact-filters ... 1.4 -> 3.0.1
org.apache.maven.shared:maven-dependency-tree ........... 2.2 -> 3.0.1
org.apache.maven.wagon:wagon-file ...................... 2.10 -> 3.0.0
org.apache.maven.wagon:wagon-http-lightweight .......... 2.10 -> 3.0.0
org.apache.maven.wagon:wagon-provider-api .............. 2.10 -> 3.0.0
org.apache.maven.wagon:wagon-ssh ....................... 2.10 -> 3.0.0
org.codehaus.plexus:plexus-container-default ... 1.0-alpha-9-stable-1 -> 1.7.1
org.codehaus.plexus:plexus-utils ..................... 3.0.24 -> 3.1.0

The following dependencies in pluginManagement of plugins have newer versions:
org.apache.maven.doxia:doxia-core ......................... 1.2 -> 1.8
org.apache.maven.shared:maven-shared-resources ................ 1 -> 2

The following dependencies in Plugin Dependencies have newer versions:
org.apache.maven.doxia:doxia-core ......................... 1.2 -> 1.8
org.codehaus.mojo:extra-enforcer-rules ...... 1.0-beta-3 -> 1.0-beta-7
```

---

**Affects:** 3.0.0

**Issue Links:**
- [MPIR-364](https://issues.apache.org/jira/browse/MPIR-364) Update plexus-container-default to 1.7.1 causes tests with maven-plugin-testing-harness 1.3 to fail
 (_**"is blocked by"**_)
- [MPIR-361](https://issues.apache.org/jira/browse/MPIR-361) Upgrade parent to 31
 (_**"is blocked by"**_)
- [MPIR-365](https://issues.apache.org/jira/browse/MPIR-365) Failed tests on Java 9 due to error with keytool plugin "keytool: not found"

- [MPIR-345](https://issues.apache.org/jira/browse/MPIR-345) Upgrade to Doxia 1.8

- [MPIR-346](https://issues.apache.org/jira/browse/MPIR-346) Upgrade to Doxia Sitetools 1.8.1

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

I would check this after MPIR-361 is done...apart from that a model upgrade to 3.5.3 is not a good idea...

**Comment by jira-importer** [comment]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** commented

Yes, lets keep this as a discussion and decide what plugins needs to be updated to which version then i will prepare PR for each update. 

Bloker is MPIR-364 and I dont know how to fix this

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Hi, could we do such discussions on the dev list or via dev chat ? Apart from that I would start with the current failing build...

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

So, I have pushed a branch with some updates which do not require any actions. These

```
maven-dependency-tree
aether-connector-wagon
maven-plugin-testing-harness
```

require changes. I do not even know wether they can be updated at all.

[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte), can they because you haven't done this with the Maven 3 upgrade.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

maven-dependency-tree: with 3.0 the only the graph-based implementation exists, but its result is different to the tree-based implementation, weird enough. Might be a bug in maven-dependency-tree.
aether-connector-wagon: IIRC it is only required for testing, I want to use the one matching Maven 3.0 (or close to it)
maven-plugin-testing-harness: the selected version is the one compatible with Maven 3.0(.x)

So the versions of these dependencies are correct.

**Comment by jira-importer** [comment]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** commented

From test dependences httpunit:httpunit can be updated to org.http.unit:httpunit:1.7.2
In general httpunit looks is not maintained now - later, as alternative i think JSoup would be good to consider

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte), thanks for the update. I will leave them as-is in the branch and merge to master before I run the release.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [ce8eb56f7c6a88902d0dca5275a7c73e3ee805b4](https://gitbox.apache.org/repos/asf?p=maven-project-info-reports-plugin.git;a=commit;h=ce8eb56f7c6a88902d0dca5275a7c73e3ee805b4).
